lib/src/main/java/com/auth0/spring/security/api/JwtAuthenticationEntryPoint.java

```diff
@@ -9,8 +9,22 @@
 import java.io.IOException;
 
 public class JwtAuthenticationEntryPoint implements AuthenticationEntryPoint {
+
+    final String audience;
+    final String issuer;
+
+    public JwtAuthenticationEntryPoint(String audience, String issuer) {
+        this.audience = audience;
+        this.issuer = issuer;
+    }
+
     @Override
     public void commence(HttpServletRequest request, HttpServletResponse response, AuthenticationException authException) throws IOException, ServletException {
+        response.addHeader(
+                "WWW-Authenticate",
+                String.format("Bearer realm=\"%s\", authorization_uri=\"%soauth/token\"", this.audience, this.issuer)
+        );
+
```

Comment:
Let's use `HttpHeaders.WWW-Authenticate` instead of hard-coding the String literal.

For the header value, the `authorization_uri` you included here is really the token endpoint; the authorization uri would be `/authorize`. As the specification does not require the `authorization_uri`, I think it would be ok to not include it unless you have a use case you are thinking of.

Reply:
Hey Jimmy, thanks for your feedback.
The replacement of the headers key by the constant makes absolutely sense.

Regarding to the authorization_uri, the removal of this attribute would change the constructor to `JwtAuthenticationEntryPoint(String audience)`. What do you think of keeping the current constructor `JwtAuthenticationEntryPoint(String audience, String issuer)` as a secondary one and making `authorization_uri` depending on the `issuer` so that `authorization_uri` is only written when the `issuer` is set?
I don't have such a deep understanding of the standard, as this is not my normal business. So feel free to reject my suggestions in case you see the standard violated here ;)
...and of course I will change the authroization_uri to `[issuer]/authorize`

Reply:
Making authroization_uri dependent on the issuer was a fallacy. I will provide another solution soon. thx.